src/test/java/com/gooddata/warehouse/WarehouseServiceIT.java

```diff
@@ -52,12 +52,14 @@ public void shouldCreateWarehouse() throws Exception {
         onRequest()
                 .havingMethodEqualTo("POST")
                 .havingPathEqualTo(Warehouses.URI)
+                .havingHeader(GoodData.GDC_HOST_HEADER)
```

Comment:
Well, it's enough to test the presence of the header only in one test (to keep other tests as concise as possible). On the other hand the test for the absence of the header when null is used is missing.